sale_financial_risk_pos_compatibility/static/src/js/CreateOrderPopup.esm.js

```diff
@@ -0,0 +1,89 @@
+/** @odoo-module **/
+
+import CreateOrderPopup from "point_of_sale.CreateOrderPopup";
+import Registries from "point_of_sale.Registries";
+import framework from "web.framework";
+
+export const CreateOrderPopupRisk = (CreateOrderPopup) =>
+    class CreateOrderPopupRisk extends CreateOrderPopup {
+        async _actionCreateSaleOrder(order_state) {
+            if (order_state === "draft") {
+                // Default behavior
+                return await super._actionCreateSaleOrder(order_state);
+            }
+            this.extraContext = false;
+            const order = this.env.pos.get_order();
+            const risk_amount =
+                order.get_total_with_tax() + order.get_rounding_applied();
+            const [partner] = await this.rpc({
+                model: "res.partner",
+                method: "read",
+                args: [
+                    order.partner.id,
+                    [
+                        "risk_exception",
+                        "risk_sale_order_limit",
+                        "risk_sale_order",
+                        "risk_sale_order_include",
+                        "risk_total",
+                        "credit_limit",
+                    ],
+                ],
+            });
+            let exception_msg = "";
+            if (partner.risk_exception) {
+                exception_msg = this.env._t("Financial risk exceeded.\n");
+            } else if (
+                partner.risk_sale_order_limit &&
+                partner.risk_sale_order + risk_amount > partner.risk_sale_order_limit
+            ) {
+                exception_msg = this.env._t(
+                    "This sale order exceeds the sales orders risk.\n"
+                );
+            } else if (
+                partner.risk_sale_order_include &&
+                partner.risk_total + risk_amount > partner.credit_limit
+            ) {
+                exception_msg = this.env._t(
+                    "This sale order exceeds the financial risk.\n"
+                );
+            } else {
+                return await super._actionCreateSaleOrder(order_state);
+            }
+            const {confirmed} = await this.showPopup("ConfirmRiskPopup", {
```

Comment:
it seems to be necessary to show different popup depending on the settings
```js
            if (this.env.pos.user.has_role_risk_manager) {
                const {confirmed} = await  this.showPopup("ConfirmPopup", {
                    title: this.env._t("Partner risk exceeded"),
                    body: exception_msg,
                });
                if (confirmed) {
                    return await super._actionCreateSaleOrder(order_state);
                } 
            } else {
                await this.showPopup("ErrorPopup", {
                    title: this.env._t("Partner risk exceeded"),
                    body: exception_msg,
                });
            }
            return await this.cancel();
```

creating a new popup is unnecessary

Reply:
Resolved